### PR TITLE
feat(constants): Add utility functions for config access

### PR DESCRIPTION
**Pull request by Jacques-Murray**
This PR adds utility functions to better encapsulate configuration access in constants.py:

- Add [get_env_vars()](cci:1://file:///c:/Users/jmmur/source/repos/Jacques-Murray/crewAI/src/crewai/cli/constants.py:169:0-181:19) to access provider environment variables
- Add [get_providers()](cci:1://file:///c:/Users/jmmur/source/repos/Jacques-Murray/crewAI/src/crewai/cli/constants.py:184:0-191:20) to retrieve supported providers list
- Add [get_models()](cci:1://file:///c:/Users/jmmur/source/repos/Jacques-Murray/crewAI/src/crewai/cli/constants.py:194:0-206:17) to access provider model configurations

These functions improve code maintainability by:
- Providing a clean interface to access configuration data
- Adding type hints and documentation for better code clarity
- Enabling filtered access to specific provider configurations

### CODE REVIEW

**Review by joaovictorcamargo — APPROVED** [review]

**Comment by joaomdmoura** [comment]
This is a review from a crew. I'm testing a crew for PRs, so if it sounds weird that is the reason, you are one of the frist ones to see this. :) I'll try to personally look into it as well

## Specific Code Improvements
1. **Add Type Hints**: Introducing Python type hints will improve code clarity and maintainability. For instance:
    ```python
    from typing import Dict, List, Optional

    def get_env_vars(provider: Optional[str] = None) -> Dict[str, List[str]]:
    ```

2. **Enhance Input Validation**: Ensure that the `provider` parameter is validated to prevent erroneous inputs. Recommended changes:
    ```python
    def get_env_vars(provider: Optional[str] = None) -> Dict[str, List[str]]:
        if provider is not None and not isinstance(provider, str):
            raise TypeError("Provider must be a string or None")
        if provider and provider not in ENV_VARS:
            raise ValueError(f"Unknown provider: {provider}")
        return ENV_VARS[provider] if provider else ENV_VARS
    ```

3. **Normalize Return Types**: The `get_models()` function should return a consistent type regardless of the input:
    ```python
    def get_models(provider: Optional[str] = None) -> Dict[str, List[str]]:
        if provider and provider in MODELS:
            return {provider: MODELS[provider]}
        return MODELS
    ```

4. **Organize Code Structure**: Consider grouping related constants with the utility functions to enhance readability:
    ```python
    # Provider-related constants and utility functions here
    ```

5. **Add Unit Tests**: It is essential to cover the new functions with unit tests to ensure they function as expected. Example tests might include:
    ```python
    def test_get_env_vars():
        assert get_env_vars() is not None
        assert isinstance(get_env_vars("openai"), list)
        with pytest.raises(ValueError):
            get_env_vars("invalid_provider")
    ```

## Historical Context from Related PRs
This PR builds upon prior discussions in related pull requests, particularly the following:

- **[PR #1552: Brandon/cre 402 add input into flow kickoff](https://github.com/crewAIInc/crewAI/pull/1552)**: Focuses on enhancing user interactions, laying the groundwork for the input handling seen in this PR.

- **[PR #1430: feat/updated CLI to allow for model selection & submitting API keys](https://github.com/crewAIInc/crewAI/pull/1430)**: Highlights the importance of user-driven choices in CLI functionality, which resonates with the current utility functions added for provider and model access.

These pull requests collectively indicate a strong trend towards improving user experience and security in CLI interactions, emphasizing the necessity of sound configuration management.

## Implications for Related Files
Adding these utility functions enhances the encapsulation of configuration management, reducing direct access to raw data structures. It will allow changes in the underlying configuration constants without impacting other parts of the code, provided that the interface remains consistent. This approach not only improves maintainability but also encourages collaborative development by fostering a better understanding of data flow.

In conclusion, while the current implementation is a solid foundation, implementing the suggested improvements will align the code with best practices and ensure a more robust, maintainable system moving forward.

**Review by joaovictorcamargo — APPROVED** [review]

**Comment by bhancockio** [comment]
Hey @Jacques-Murray! I like where you are going with this. 

However, where in the code should we call these updated functions?

We try not to add code to the library that doesn't get called by anything.
